indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentTransactionalAppendAction.java

```diff
@@ -148,65 +146,58 @@ public SegmentPublishResult perform(Task task, TaskActionToolbox toolbox)
           task.getType()
       );
     }
-    // Verify that all the locks are of expected type
-    final List<TaskLock> locks = toolbox.getTaskLockbox().findLocksForTask(task);
-    for (TaskLock lock : locks) {
-      if (lock.getType() != TaskLockType.APPEND) {
-        throw InvalidInput.exception(
-            "Cannot use action[%s] for task[%s] as it is holding a lock of type[%s] instead of [APPEND].",
-            "SegmentTransactionalAppendAction", task.getId(), lock.getType()
+    final TaskLockbox taskLockbox = toolbox.getTaskLockbox();
+    TaskLocks.checkLockCoversSegments(task, taskLockbox, segments);
+    taskLockbox.acquireTransactionalAppendLock(task, TaskLockbox.LOCK_ACQUIRE_TIMEOUT_MILLIS);
+    try {
+      final String datasource = task.getDataSource();
+      final Map<DataSegment, ReplaceTaskLock> segmentToReplaceLock
+          = TaskLocks.findReplaceLocksCoveringSegments(datasource, toolbox.getTaskLockbox(), segments);
+
+      final CriticalAction.Action<SegmentPublishResult> publishAction;
+      final String taskAllocatorId = ((PendingSegmentAllocatingTask) task).getTaskAllocatorId();
+      if (startMetadata == null) {
+        publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegments(
+            segments,
+            segmentToReplaceLock,
+            taskAllocatorId,
+            segmentSchemaMapping
+        );
+      } else {
+        publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegmentsAndMetadata(
+            segments,
+            segmentToReplaceLock,
+            startMetadata,
+            endMetadata,
+            taskAllocatorId,
+            segmentSchemaMapping
         );
       }
-    }
-
-    TaskLocks.checkLockCoversSegments(task, toolbox.getTaskLockbox(), segments);
 
-    final String datasource = task.getDataSource();
-    final Map<DataSegment, ReplaceTaskLock> segmentToReplaceLock
-        = TaskLocks.findReplaceLocksCoveringSegments(datasource, toolbox.getTaskLockbox(), segments);
-
-    final CriticalAction.Action<SegmentPublishResult> publishAction;
-    final String taskAllocatorId = ((PendingSegmentAllocatingTask) task).getTaskAllocatorId();
-    if (startMetadata == null) {
-      publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegments(
-          segments,
-          segmentToReplaceLock,
-          taskAllocatorId,
-          segmentSchemaMapping
-      );
-    } else {
-      publishAction = () -> toolbox.getIndexerMetadataStorageCoordinator().commitAppendSegmentsAndMetadata(
-          segments,
-          segmentToReplaceLock,
-          startMetadata,
-          endMetadata,
-          taskAllocatorId,
-          segmentSchemaMapping
-      );
-    }
-
-    final SegmentPublishResult retVal;
-    try {
+      final SegmentPublishResult retVal;
```

Comment:
Declaration and assignment can be in a single line.

Reply:
Rename this to `publishResult`.

Reply:
Done